telemetry/specs/INT.mdk

```diff
@@ -621,19 +621,190 @@ corruption at preceding hops.
 
-1. *INT over TCP/UDP* - A shim header is inserted following TCP/UDP
+1. *INT over IPv4/GRE* - INT Headers are carried in the GRE encapsulation.
+2. *INT over IPv6* - INT Headers are carried in the IPv6 packets as Hop-by-Hop option.
+3. *INT over TCP/UDP* - A shim header is inserted following TCP/UDP
 header. INT Headers are carried between this shim header and TCP/UDP payload.
 This approach doesn’t rely on any tunneling/virtualization mechanism and is
 versatile to apply INT to both native and virtualized traffic.
-2. *INT over VXLAN* - VXLAN generic protocol extensions
+4. *INT over VXLAN* - VXLAN generic protocol extensions
 (draft-ietf-nvo3-vxlan-gpe) are used to carry INT Headers between
 the VXLAN header and the encapsulated VXLAN payload.
-3. *INT over Geneve* - Geneve is an extensible tunneling framework, allowing
+5. *INT over Geneve* - Geneve is an extensible tunneling framework, allowing
 Geneve options to be defined for INT Headers.
 
+### INT over IPv4/GRE
+
+In case the traffic being monitored is not encapsulated by any virtualization
+header, INT over VXLAN or INT over Geneve is not helpful. Instead, 
+the INT metadata can be carried in the GRE header. 
+
+IPv4 GRE Option format for carrying INT Header and 
+Metadata:
+
+When the INT data fields are included within an encapsulation that identifies 
+the next protocol using an EtherType (e.g., GRE or Geneve) the presence of 
+INT data fields are identified with TBD_INT. When this EtherType is used, 
+an additional INT header is also included. This header indicates the type 
+of INT data that follows, and the next protocol that follows the INT data.
+
+0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+|C|R|K|S|s|Recur|   Flags | Ver |   Protocol Type = TBD_INT     |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  |
+|     Checksum (optional)       |        Offset (Optional)      |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  G
+|                         Key (Optional)                        |  R
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  E
+|                   Sequence Number (Optional)                  |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  |
+|                        Routing (Optional)                     |  |     
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
+|    INT-Type   |     Length    |        Next Protocol          |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+  I
+|                                                               |  N
+|  Variable Option Data (INT Metadata Headers and Metadata]     |  T                     
+|                                                               |  |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+<-+
```

Comment:
We need to define the INT-Type, Length, and Next Protocol fields, as we do for the other transports.

Next Protocol: 16 bits Next Protocol field contains the EtherType value that specifies the protocol type of the next header following the INT headers.

We have a bunch of text in VXLAN GPE talking about the possibility of multiple INT headers, each specified in the previous header's next protocol, with the last header indicating a non-INT value. Not sure whether we should have any of that here.

Reply:
Done.